lib/python/rose/macro.py

```diff
@@ -246,8 +246,8 @@ def _get_config_sections(self, config_data):
             for key, node in config_data.value.items():
                 if isinstance(node.value, dict):
                     sections.append(key)
-                elif "" not in sections:
```

Comment:
I thought the `root` section always exists?
* It exists but iis empty for an empty file.
* Otherwise, it contains section nodes even if it does not have string nodes.

(Getting philosophical here.)